languages/javascript/generic/js_to_generic.ml

```diff
@@ -223,60 +223,65 @@ and literal x : G.literal =
       G.Regexp (v1, v2)
 
 and expr (x : expr) =
-  (match x with
+  match x with
   | ObjAccessEllipsis (v1, v2) ->
       let v1 = expr v1 in
-      G.DotAccessEllipsis (v1, v2)
+      G.DotAccessEllipsis (v1, v2) |> G.e
   (* not sure this is actually a valid JS/TS construct *)
   | Cast (v1, v2, v3) ->
       let v1 = expr v1 in
       let v3 = type_ v3 in
-      G.Cast (v3, v2, v1)
+      G.Cast (v3, v2, v1) |> G.e
   (* converting to Cast as it's mostly the same *)
   | TypeAssert (v1, v2, v3) ->
       let v1 = expr v1 in
       let v3 = type_ v3 in
-      G.Cast (v3, v2, v1)
+      G.Cast (v3, v2, v1) |> G.e
   | ExprTodo (v1, v2) ->
       let v2 = list expr v2 in
-      G.OtherExpr (v1, v2 |> Common.map (fun e -> G.E e))
-  | L x -> G.L (literal x)
+      G.OtherExpr (v1, v2 |> Common.map (fun e -> G.E e)) |> G.e
+  | ParenExpr (l, e, r) ->
+      let e = expr e in
+      H.set_e_range l r e;
+      e
```

Comment:
Out of curiosity, why did you choose this rather than translating to an AST_generic.ParenExpr?

Reply:
See #7437 for a more detailed explanation, but basically when I tried that with Python it broke a whole bunch of stuff

Reply:
We actually should probably delete ParenExpr from the AST and write to `e_range` instead for the one spot it's actually used